src/gleam/list.gleam

```diff
@@ -454,6 +454,64 @@ pub fn fold_right(list: List(a), from initial: b, with fun: fn(a, b) -> b) -> b
   }
 }
 
+/// Like fold but the folding function also receives the index of the current element.
+///
+/// ## Examples
+///
+/// ```
+/// ["a", "b", "c"]
+/// |> list.index_fold([], fn(index, item, acc) { ... })
+/// ```
```

Comment:
I found the double indentation quite difficult to get right. It is very finicky.
Then I found that using ``` works fine for code highlighting, which is easier to use.